src/main/java/me/yukun99/ip/core/TaskFinder.java

```diff
@@ -0,0 +1,153 @@
+package me.yukun99.ip.core;
+
+import me.yukun99.ip.tasks.Task;
+import me.yukun99.ip.tasks.ToDo;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Handles searching of tasks by keyword.
+ */
+public class TaskFinder {
+	private final HashMap<String, List<Task>> wordTaskMap = new HashMap<>();
+
+	/**
+	 * Adds a new task to the TaskFinder instance.
+	 *
+	 * @param task Task to be added to the TaskFinder instance.
+	 * @param name Name of task to be added.
+	 */
+	public void addTask(Task task, String name) {
+		for (String word : parseWords(name)) {
+			List<Task> tasks = new ArrayList<>();
+			if (wordTaskMap.containsKey(word)) {
+				tasks = wordTaskMap.get(word);
+			}
+			tasks.add(task);
+			wordTaskMap.put(word, tasks);
+		}
+	}
+
+	/**
+	 * Deletes a task from the TaskFinder instance.
+	 *
+	 * @param task Task to be deleted from the TaskFinder instance.
+	 * @param name Name of task to be deleted.
+	 */
+	public void deleteTask(Task task, String name) {
+		for (String word : parseWords(name)) {
+			if (!wordTaskMap.containsKey(word)) {
+				continue;
+			}
+			List<Task> tasks = wordTaskMap.get(word);
+			tasks.remove(task);
+			wordTaskMap.put(word, tasks);
+		}
+	}
+
+	/**
+	 * Parses the words in a task name.
+	 *
+	 * @param name Task name to be parsed.
+	 * @return List containing all the words in the task name.
+	 */
+	private List<String> parseWords(String name) {
```

Comment:
Maybe the name could be changed to parsedWords to make it sound more like a noun?

Reply:
Functions are supposed to be named as verbs I think